exchange/events.go

```diff
@@ -0,0 +1,89 @@
+package exchange
+
+import (
+	"encoding/json"
+	"time"
+
+	jsonpatch "github.com/evanphx/json-patch"
+	"github.com/mxmCherry/openrtb"
+	"github.com/prebid/prebid-server/adapters"
+	"github.com/prebid/prebid-server/analytics"
+	"github.com/prebid/prebid-server/config"
+	"github.com/prebid/prebid-server/endpoints/events"
+	"github.com/prebid/prebid-server/openrtb_ext"
+	"github.com/prebid/prebid-server/pbsmetrics"
+)
+
+// eventsData has configuration fields needed for adding event tracking to an auction response
+type eventsData struct {
+	accountID          string
+	enabledForAccount  bool
+	enabledForRequest  bool
+	auctionTimestampMs int64
+	integration        pbsmetrics.DemandSource // web app amp
+	bidderInfos        adapters.BidderInfos
+	externalURL        string
+}
+
+// getExtEventsData creates an eventsData object from the different configuration sources
+func getExtEventsData(requestExtPrebid *openrtb_ext.ExtRequestPrebid, ts time.Time, account *config.Account, bidderInfos adapters.BidderInfos, externalURL string) *eventsData {
+	return &eventsData{
+		accountID:          account.ID,
+		enabledForAccount:  account.EventsEnabled,
+		enabledForRequest:  requestExtPrebid != nil && requestExtPrebid.Events != nil, // || account.analytics.auction-events.<web|app|amp>
```

Comment:
What's with the comment here?

Reply:
This should be on the account line. It's a note I left regarding the algorithm making a mention to a channel event configuration for account, which I decided to leave out for now, to add in the next iteration.

> A new account-level configuration is proposed to define whether an account needs server-side events by channel. e.g account 111 gets AMP/App events, but not web events. Account 222 gets events for all 3 channels: Web/AMP/App.
If an account doesn't specify the channel config, the default is that AMP and App are "true" if events are enabled.

Reply:
Gotcha. Could you perhaps leave this out or better explain it? It looks like a mistake without that context.